app/main.py

```diff
@@ -1,2 +1,97 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.storage_length = 8
+        self.storage = [None] * self.storage_length
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        new_item = (key, hash(key), value)
+        index = self.get_index(new_item)
+        self.storage[index] = new_item
+        if len([i for i in self.storage if i]) >= 2 * self.storage_length / 3:
+            self.resize()
+
+    def get_index(self, new_item: tuple) -> int:
+        index = new_item[1] % self.storage_length
+        for item_index, item in enumerate(self.storage):
+            if item:
+                if item[0] == new_item[0] and item[1] == new_item[1]:
+                    return item_index
+        while True:
+            if not self.storage[index]:
+                return index
+            index = (index + 1) % self.storage_length
+
+    def resize(self) -> None:
+        temp_storage = [i for i in self.storage if i]
+        self.storage_length *= 2
+        self.storage = [None] * self.storage_length
+        for i in temp_storage:
+            self.__setitem__(i[0], i[2])
+        del temp_storage
+
+    def __getitem__(self, item: Any) -> Any:
+        key_hash = hash(item)
+        for item_ in self.storage:
```

Comment:
you should find by key, not by item

Reply:
i just variable, that stores key value named as "item". The logic of code - looking by key